docs/changes.rst

```diff
@@ -22,6 +22,18 @@ You should generally assume that an API is internal unless you have specific
 information to the contrary.
 
 ------------------
+3.9.0 - 2017-05-18
+------------------
+
+Hypothesis now uses ``inspect.getfullargspec`` internally.
+On Python 2, there are no visible changes.
+
+On Python 3, function annotations are no longer discarded by various
+decorators.  Keyword-only arguments are no longer silently discarded, leading
```

Comment:
Personally I might split this out as follows:

* `@given` and `@composite` now preserve annotations on arguments.
* `@composite` now supports keyword only arguments.

In particular this spells out that keyword only arguments still don't work with `@given` (which I don't propose you try to fix in this PR - I think it's probably moderately annoying to do and I don't think it's a big deal).